### PR TITLE
PP-863 Updated Charge Api Resource endpoint to allow searching by `card_brand`

### PR DESCRIPTION
**Pull request by ckalista**
## WHAT

Updated Charge Api Resource endpoint  and `ChargeDao` to allow searching by `card_brand`.

### CODE REVIEW

**Comment by ckalista** [comment]
rebuild

File: src/main/java/uk/gov/pay/connector/dao/ChargeDao.java [hunk 1]
```diff
@@ -135,6 +136,8 @@ public ChargeEntity mergeAndNotifyStatusHasChanged(ChargeEntity chargeEntity) {
             predicates.add(likePredicate(cb, charge.get(EMAIL), params.getEmail()));
         if (params.getChargeStatuses() != null && !params.getChargeStatuses().isEmpty())
             predicates.add(charge.get(STATUS).in(params.getChargeStatuses()));
+        if (params.getCardBrand() != null && !params.getCardBrand().isEmpty())
```
Comment (imranbohoran) [inline src/main/java/uk/gov/pay/connector/dao/ChargeDao.java]: Shouldn't this be a simple `StringUtils.isNotBlank(params.getCardBrand())` as with other cases.

File: src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoITest.java [hunk 2]
```diff
@@ -560,6 +584,34 @@ public void searchChargesShouldBeOrderedByCreationDateDescending() {
...
+        // then
+        assertThat(charges.size(), is(1));
+        ChargeEntity charge = charges.get(0);
+
+        assertThat(charge.getId(), is(defaultTestCharge.getChargeId()));
+        assertThat(charge.getAmount(), is(defaultTestCharge.getAmount()));
+        assertThat(charge.getReference(), is(defaultTestCharge.getReference()));
+        assertThat(charge.getDescription(), is(DESCRIPTION));
+        assertThat(charge.getStatus(), is(defaultTestCharge.getChargeStatus().toString()));
+        assertDateMatch(charge.getCreatedDate().toString());
```
Comment (imranbohoran) [inline src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoITest.java]: As we are asserting other things is there are reason why we are not asserting the brand for the charge we get back?

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 3]
```diff
@@ -690,7 +710,7 @@ private void assertResultsAndJustSelfLinkWhenJustOneResult() {
                 .getTransactions()
                 .statusCode(OK.getStatusCode())
                 .contentType(JSON)
-                // then no prev and next link
+                        // then no prev and next link
```
Comment (imranbohoran) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: minor: tabs perhaps. I'm not sure if we need the comment though

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 4]
```diff
@@ -714,7 +734,7 @@ private void assertResultsWhenPageAndDisplaySizeNotSet() {
...
```
Comment (imranbohoran) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: minor: tabs perhaps. I'm not sure if we need the comment though

**Review by imranbohoran — COMMENTED** [review]
Review done. Few minor comments added.

File: src/main/java/uk/gov/pay/connector/dao/ChargeDao.java (hunk 1 above)
Comment (ckalista) [reply src/main/java/uk/gov/pay/connector/dao/ChargeDao.java]: Indeed. Fixed.

File: src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoITest.java (hunk 2 above)
Comment (ckalista) [reply src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoITest.java]: Added missing assertions on brand type to all relevant tests.

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java (hunk 3 above)
Comment (ckalista) [reply src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: Removed the comment.

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java (hunk 4 above)
Comment (ckalista) [reply src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: Removed the comment.

**Comment by ckalista** [comment]
rebuild

**Comment by ckalista** [comment]
rebuild

**Review by imranbohoran — APPROVED** [review]
Approved

**Pull request merged**